src/containers/NFT/NFTHeader/index.js

```diff
@@ -0,0 +1,219 @@
+import React, { Component, useEffect, useContext, useState } from 'react';
+import { useTranslation, withTranslation } from 'react-i18next';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import { loadNFTState } from './actions';
+import Loader from '../../shared/components/Loader';
+import '../../shared/css/nested-menu.css';
+import './styles.css';
+import SocketContext from '../../shared/SocketContext';
+import Tooltip from '../../shared/components/Tooltip';
+import Copy from '../../shared/components/Copy';
+
+const NFTHeader = props => {
+  const { t } = useTranslation();
+  const { actions, tokenId, data, loading, language } = props;
+  const rippledSocket = useContext(SocketContext);
+  const [tooltip, setTooltip] = useState(null);
+
+  useEffect(() => {
+    actions.loadNFTState(tokenId, rippledSocket.clioSocket);
+  }, [tokenId]);
+
+  const showTooltip = (event, d) => {
+    setTooltip({ ...d, mode: 'nftId', x: event.pageX, y: event.pageY });
+  };
+
+  const hideTooltip = () => {
+    setTooltip(null);
+  };
+
+  const renderDetails = () => {
+    const { minted, domain, emailHash, NFTTaxon, uri, transferFee } = data;
+    const abbrvEmail = emailHash?.replace(/(.{20})..+/, '$1...');
```

Comment:
again i'm not sure this regex is right. also i think it's easier to read like so in this case:

```javascript
const abbrvEmail = emailHash?.slice(0, 20).concat('...')
```